packages/tc-schema-sdk/data-accessors/type.js

```diff
@@ -133,25 +133,53 @@ const getType = function(
 	}
 
 	if (withRelationships) {
-		Object.entries(typeSchema.properties).forEach(([propName, def]) => {
-			if (def.relationship || def.cypher) {
-				if (def.hidden) {
-					delete typeSchema.properties[propName];
+		Object.keys(typeSchema.properties).forEach(key => {
+			let definition;
+			try {
+				definition = this.getRelationshipType(typeSchema.name, key, {
+					primitiveTypes,
+					includeMetaFields,
+				});
+			} catch (error) {
+				definition = typeSchema.properties[key];
+			}
+
+			if (definition.relationship || definition.cypher) {
+				if (definition.hidden) {
+					delete typeSchema.properties[key];
+					return;
 				}
-				Object.assign(def, {
-					hasMany: def.hasMany || false,
-					isRelationship: !!(def.relationship || def.cypher),
+				typeSchema.properties[key] = {
+					...definition,
+					hasMany: definition.hasMany || false,
+					isRelationship: !!(
+						definition.relationship || definition.cypher
+					),
 					writeInactive:
-						'writeInactive' in def ? def.writeInactive : false,
+						'writeInactive' in definition
+							? definition.writeInactive
+							: false,
 					showInactive:
-						'showInactive' in def ? def.showInactive : true,
-				});
+						'showInactive' in definition
+							? definition.showInactive
+							: true,
+				};
 			}
 		});
 	} else {
-		Object.entries(typeSchema.properties).forEach(([propName, def]) => {
-			if (def.relationship) {
-				delete typeSchema.properties[propName];
+		Object.keys(typeSchema.properties).forEach(key => {
+			let definition;
+			try {
+				definition = this.getRelationshipType(typeSchema.name, key, {
```

Comment:
here you are using it to check if a property is a relationship. Is it worth implementing `isRelationship` which (once you've split realtionshipTypes into a separate array to types in the rawData) can just check to see if it can find the relationship type in the array?

Reply:
Yes - current this data-acceccor is working based on combined types array, so when we move relationship types toward other root property, this logic can be simpler (maybe we can check without calling `this.getRelationshipType`)